integration/tests/access/access_test.go

```diff
@@ -112,3 +118,85 @@ func (s *AccessSuite) TestAPIsAvailable() {
 		assert.NoError(t, err, "failed to ping access node")
 	})
 }
+
+// TestSignerIndicesDecoding tests that access node uses signer indices' decoder to correctly parse encoded data in blocks.
+// This test receives blocks from consensus follower and then requests same blocks from access API and checks if returned data
+// matches.
+func (s *AccessSuite) TestSignerIndicesDecoding() {
+
+	container := s.net.ContainerByName(testnet.PrimaryAN)
+
+	ctx, cancel := context.WithCancel(s.ctx)
+	defer cancel()
+
+	// create access API
+	grpcAddress := container.Addr(testnet.GRPCPort)
+	conn, err := grpc.DialContext(ctx, grpcAddress, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	require.NoError(s.T(), err, "failed to connect to access node")
+	defer conn.Close()
+
+	client := accessproto.NewAccessAPIClient(conn)
+
+	// query latest finalized block
+	latestFinalizedBlock, err := makeApiRequest(client.GetLatestBlockHeader, ctx, &accessproto.GetLatestBlockHeaderRequest{
+		IsSealed: false,
+	})
+	require.NoError(s.T(), err)
+
+	blockByID, err := makeApiRequest(client.GetBlockHeaderByID, ctx, &accessproto.GetBlockHeaderByIDRequest{Id: latestFinalizedBlock.Block.Id})
+	require.NoError(s.T(), err)
+
+	require.Equal(s.T(), latestFinalizedBlock, blockByID, "expect to receive same block by ID")
+
+	blockByHeight, err := makeApiRequest(client.GetBlockHeaderByHeight, ctx,
+		&accessproto.GetBlockHeaderByHeightRequest{Height: latestFinalizedBlock.Block.Height})
+	require.NoError(s.T(), err)
+
+	require.Equal(s.T(), blockByID, blockByHeight, "expect to receive same block by height")
+
+	// stop container, so we can access it's state and perform assertions
+	err = s.net.StopContainerByName(ctx, testnet.PrimaryAN)
+	require.NoError(s.T(), err)
+
+	err = container.WaitForContainerStopped(5 * time.Second)
+	require.NoError(s.T(), err)
+
+	// open state to build a block singer decoder
+	state, err := container.OpenState()
+	require.NoError(s.T(), err)
+
+	// create committee so we can create decoder to assert validity of data
+	committee, err := committees.NewConsensusCommittee(state, container.Config.NodeID)
+	require.NoError(s.T(), err)
+	blockSignerDecoder := signature.NewBlockSignerDecoder(committee)
+	// checks if
+	assertSignerIndicesValidity := func(msg *entities.BlockHeader) {
+		block, err := convert.MessageToBlockHeader(msg)
+		require.NoError(s.T(), err)
+		decodedIdentities, err := blockSignerDecoder.DecodeSignerIDs(block)
+		require.NoError(s.T(), err)
+		// transform to assert
+		var transformed [][]byte
+		for _, identity := range decodedIdentities {
+			identity := identity
+			transformed = append(transformed, identity[:])
+		}
+		assert.ElementsMatch(s.T(), transformed, msg.ParentVoterIds, "response must contain correctly encoded signer IDs")
+	}
```

Comment:
I'm wondering why this function is declared inline here. We don't save space, since the function is only used once, but the logic isn't organized in execution order.

I would prefer to move the signer indices checking logic inline (line 192)? Or move this to a static (not inline) function.

(Please also complete or remove the comment line `// checks if`)